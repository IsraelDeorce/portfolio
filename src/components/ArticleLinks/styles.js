import styled from 'styled-components'

export const ArticleLinks = styled.ul`
  display: flex;
  flex-direction: column;
  padding-left: 1em;
  & > * {
    color: ${({ theme }) => theme.c_brackets};
    &:hover {
      color: ${({ theme }) => theme.c_varType};
    }
    padding-bottom: 1em;
  }
`
