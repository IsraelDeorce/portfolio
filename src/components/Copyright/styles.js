import styled from 'styled-components'

export const Copyright = styled.div`
  display: flex;
  justify-content: center;
`
