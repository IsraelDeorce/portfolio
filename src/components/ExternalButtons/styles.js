import styled from 'styled-components'

export const ExternalButtons = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 20px;
  font-size: 1.25rem;
`
