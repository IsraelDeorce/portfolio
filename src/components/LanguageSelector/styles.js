import styled from 'styled-components'

export const LanguageSelector = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
  color: ${({ theme }) => theme.c_brackets};
`

export const LanguageCode = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`