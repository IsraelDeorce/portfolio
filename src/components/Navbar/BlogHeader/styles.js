import styled from 'styled-components'

export const BlogHeader = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  font-weight: 700;
  background-color: ${({ theme }) => theme.c_menu};
  color: ${({ theme }) => theme.c_brackets};
  padding: 0 1.5em;
  min-height: 60px;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (min-width: 992px) {
    display: none;
  }
`
export const Hamburguer = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
`
