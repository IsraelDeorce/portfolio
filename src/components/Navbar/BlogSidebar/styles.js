import styled from 'styled-components'

export const Background = styled.div`
  opacity: 0.7;
  background-color: ${({ theme }) => theme.c_background};
  position:fixed;
  width:100%;
  height:100%;
  top:0px;
  left:0px;
  @media (min-width: 992px) {
    display: none;
  }
`

export const BlogSidebar = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.c_menu};
  height: 100vh;
  width: 65vw;
  z-index: 2;
  box-shadow: 0 5px 30px ${({ theme }) => theme.c_tab};
  padding: 0 1em 1em 1em;
  & > * {
    width: 100%;
  }
  & > :last-child {
    align-self: center;
  }
  @media (min-width: 992px) {
    position: static;
    top: auto;
    left: auto;
    right: auto;
    height: auto;
    width: 285px;
    min-width: 285px;
    box-shadow: none;
  }
`

export const Header = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  min-height: 60px;
  border-bottom: 2px solid gray;
  & > *:not(:last-child) {
    padding: 0.35em;
  }
`

export const Preferences = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
`
