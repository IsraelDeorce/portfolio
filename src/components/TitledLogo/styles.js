import styled from 'styled-components'

export const TitledLogo = styled.div`
  display: flex;
  align-items: center;
  color: ${({ theme }) => theme.c_brackets};
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
  & > *:not(:last-child) {
    margin-right: 0.5em;
  }
`
