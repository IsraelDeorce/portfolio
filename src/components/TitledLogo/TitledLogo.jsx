import { GiDirewolf } from 'react-icons/gi'
import PropTypes from 'prop-types'
import { TEXT } from '../../utils/constants'
import * as S from './styles'

const propTypes = {
  onClick: PropTypes.func
}

function TitledLogo({ onClick }) {
  return (
    <S.TitledLogo onClick={onClick}>
      {<GiDirewolf size='32' />}
      <strong>{TEXT.MYNAME}</strong>
    </S.TitledLogo>
  )
}

TitledLogo.propTypes = propTypes

export default TitledLogo
