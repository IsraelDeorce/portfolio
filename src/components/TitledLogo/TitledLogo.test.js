import TitledLogo from './TitledLogo'
import { TEXT } from '../../utils/constants'
import * as S from './styles'

describe('<TitledLogo/>', () => {
  it('renders a direwolf icon', () => {
    expect(shallow(<TitledLogo />).find('GiDirewolf').exists()).toBe(true)
  })

  it('renders a Israel Deorce as a label', () => {
    expect(shallow(<TitledLogo />).find('strong').text()).toBe(TEXT.MYNAME)
  })

  it('calls the onCLick function when it is passed and the user clicks on the Logo', () => {
    const mockFn = jest.fn()
    shallow(<TitledLogo onClick={mockFn} />).find(S.TitledLogo).simulate('click')
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})
