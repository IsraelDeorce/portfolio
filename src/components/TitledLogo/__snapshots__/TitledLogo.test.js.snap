// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitledLogo/> Snapshot renders without crashing using shallow 1`] = `
<styled.div>
  <GiDirewolf
    size="32"
  />
  <strong>
    Israel Deorce
  </strong>
</styled.div>
`;
